homeassistant/setup.py

```diff
@@ -240,19 +242,39 @@ def log_error(msg: str) -> None:
     elif name in processed:
         return
 
-    if hasattr(module, 'DEPENDENCIES'):
+    manifest = await hass.async_add_executor_job(
+        json_util.load_json,
+        os.path.join(os.path.dirname(module.__file__), 'manifest.json')
+    )
+
+    if manifest.get('dependencies'):
```

Comment:
No need to check if manifest has the key, it's mandatory and validated by manifest validation script.